runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -47,8 +48,7 @@
  * <p>Created by mwei on 11/13/16.
  */
 @Slf4j
-public class VersionLockedObject<T> {
-
+public class VersionLockedObject<T extends ISMRObject<T>> {
```

Comment:
Nice. 

You can make `ISMRObject` closeable and then call `close` in `resetUnsafe` before the new object is constructed.

Reply:
Yep, `ICorfuSMR` is not `AutoClosable`.